### PR TITLE
Update QuickSort.java

### PR DESCRIPTION
**Pull request by manishraj27**
I've made the following changes:
1.Renamed doSort to quickSort for clarity.
2.Renamed randomPartition to partitionWithRandomPivot. 3.Added inline comments for clarity.
4.Reused existing utility methods from SortUtils.
5.Ensured error handling by checking for edge cases like null arrays. 6.Maintained the core logic of the QuickSort algorithm while optimizing readability and maintainability.

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized it.
- [ ] All filenames are in PascalCase.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [ ] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/sorts/QuickSort.java [hunk 1]
```diff
@@ -3,75 +3,77 @@
...
+     * Sorts the input array using QuickSort algorithm.
      *
-     * @param array The array to be sorted Sorts the array in increasing order
+     * @param array The array to be sorted. Sorts the array in increasing order.
+     * @return The sorted array.
      */
     @Override
     public <T extends Comparable<T>> T[] sort(T[] array) {
-        doSort(array, 0, array.length - 1);
+        quickSort(array, 0, array.length - 1);
```
Comment (TruongNhanNguyen) [inline src/main/java/com/thealgorithms/sorts/QuickSort.java:21]: __Error Handling__: You should add a check to handle cases where the array is null or has only one element, returning the array directly in such cases.
```suggestion
        if (array == null || array.length <= 1) {
            return array;
        }
        quickSort(array, 0, array.length - 1);
```

**Comment by TruongNhanNguyen** [comment]
__Random Pivot Selection__: In your implementation, the pivot selection logic to be more intuitive and clear. The pivot is always chosen from the rightmost element of the array.

File: src/main/java/com/thealgorithms/sorts/QuickSort.java [hunk 2]
```diff
@@ -3,75 +3,77 @@
...
- * @author Varun Upadhyay (https://github.com/varunu28)
- * @author Podshivalov Nikita (https://github.com/nikitap492)
+ * Implementation of the QuickSort algorithm.
+ * 
+ * @author Manish Raj (https://github.com/manishraj27)
  * @see SortAlgorithm
  */
 class QuickSort implements SortAlgorithm {
 
     /**
```
Comment (TruongNhanNguyen) [inline src/main/java/com/thealgorithms/sorts/QuickSort.java:13]:
```suggestion
    private static final int RANDOM_BOUND = 1;
    /**
```

File: src/main/java/com/thealgorithms/sorts/QuickSort.java [hunk 3]
```diff
@@ -3,75 +3,77 @@
...
-     * @param right The last index of an array
-     * @return the partition index of the array
+     * @param array The array to be partitioned.
+     * @param left  The leftmost index of the subarray.
+     * @param right The rightmost index of the subarray.
+     * @return The index of the pivot element after partitioning.
      */
-    private static <T extends Comparable<T>> int randomPartition(T[] array, int left, int right) {
+    private static <T extends Comparable<T>> int partitionWithRandomPivot(T[] array, int left, int right) {
         int randomIndex = left + (int) (Math.random() * (right - left + 1));
```
Comment (TruongNhanNguyen) [inline src/main/java/com/thealgorithms/sorts/QuickSort.java:49]:
```suggestion
        int randomIndex = left + (int) (Math.random() * (right - left + RANDOM_BOUND));
```

**Comment by TruongNhanNguyen** [comment]
__Partitioning Logic__: Simplified the partitioning logic by using a single loop instead of nested while loops, which makes it easier to understand.

**Review by BamaCharanChhandogi — COMMENTED** [review]
@manishraj27 are you still working on this PR?

**Pull request closed without merging**